syntax = "proto3";

package transport_catalogue_serialize;

message RGB {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
}

message RGBA {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
    double opacity = 4;
}

message Color {
    oneof variant {
        string string_color = 1;
        RGB RGB_color = 2;
        RGBA RGBA_color = 3;
    }
}

message RendererSettings {
    double width = 1;
    double height = 2;
    double padding = 3;
    double line_width = 4;
    double stop_radius = 5;
    int32 stop_label_font_size = 6;
    int32 bus_label_font_size = 7;
    repeated double bus_label_offset = 8;
    repeated double stop_label_offset = 9;
    Color underlayer_color = 10;
    double underlayer_width = 11;
    repeated Color color_palette = 12;
}